### DISCUSSION TITLE
Get data from a Soap Webservice

### DISCUSSION THREAD

**Discussion by Gotorus** [Budibase/budibase#8100] [Help] [answered]
Hi
I try to build a new application for my Startup. In my case I want to get some information about my parcels from a Soap Webservice of Swiss Post.

I saw, that there is only an option to import data via API Rest. Is there any possibility to do this also by a Webservice Soap? (If thats a weird question for you, I have basically not much knowledge about coding ;-).

Thanks for any tips.

**Comment by andz-bb** [comment, marked as answer]
Hi @Gotorus 

we don't currently have an integration for SOAP datasources, but we have just released Budibase 2.0 which opens up development of custom datasources and components to the community. I'm not too familiar with SOAP myself but I think it could be possible to develop a custom datasource to support it.

You can see [documentation on how to create your own datasources here](https://docs.budibase.com/docs/custom-datasource) or you can [add it as a suggestion for the community here.](https://github.com/Budibase/budibase/discussions/7807)

Alternatively it's worth checking if Swiss Post have a REST API available for your parcels. If it exists it'll be a much easier way to what you're wanting to do with Budibase!

**Reply by acseven** [reply]
One more for SOAP support, where no REST alternative is available!

**Comment by Gotorus** [comment]
Hi @andz-bb 
Thanks for your reply.
Ok, I will check these options. I placed also an request at Swiss Post. Could be that they also have a Rest API for that. But I think not.

**Comment by acseven** [comment]
@Gotorus 
SOAP datasource plugin:
https://github.com/melohagan/budibase-datasource-soap
